designer/server/src/test/scala/pl/touk/nussknacker/ui/api/NodesApiSpec.scala

```diff
@@ -0,0 +1,680 @@
+            .body(
+              equalsJson(s"""{
+                   |  "content": "\\nSamples:\\n\\n| id  | value |\\n| --- | ----- |\\n| a   | generated |\\n| b   | not existent |\\n\\nResults for a can be found [here](http://touk.pl?id=a)\\n",
+                   |  "type": "MarkdownAdditionalInfo"
+                   |}""".stripMargin)
+            )
+
+          val dataEmpty: NodeData = Enricher("1", ServiceRef("otherService", List()), "out", None)
+
+          sendNodeAsJsonAsAllpermuser(dataEmpty)
+            .when()
+            .post(s"$nuDesignerHttpAddress/api/nodes/${process.id}/additionalInfo")
+            .Then()
+            .statusCode(200)
+            .body(
+              equalTo("")
+            )
+        }
+        "return 404 for not existent process" in {
+          val data: NodeData =
+            Enricher("enricher", ServiceRef("paramService", List(Parameter("id", Expression.spel("'a'")))), "out", None)
+
+          val wrongName: String = "wrongProcessName"
+
+          sendNodeAsJsonAsAllpermuser(data)
+            .when()
+            .post(s"$nuDesignerHttpAddress/api/nodes/$wrongName/additionalInfo")
+            .Then()
+            .statusCode(404)
+            .body(
+              equalTo(s"No scenario $wrongName found")
+            )
+        }
+      }
+      "not authenticated should" - {
+        "forbid access" in {
+          createSavedProcess(process, TestCategories.Category1, TestProcessingTypes.Streaming)
+          val data: NodeData =
+            Enricher("enricher", ServiceRef("paramService", List(Parameter("id", Expression.spel("'a'")))), "out", None)
+          val json = Encoder[NodeData].apply(data)
+
+          given
+            .contentType("application/json")
+            .body(json.toString())
+            .and()
+            .auth()
+            .none
+            .when()
+            .post(s"$nuDesignerHttpAddress/api/nodes/${process.id}/additionalInfo")
+            .Then()
+            .statusCode(401)
+            .body(
+              equalTo("The resource requires authentication, which was not supplied with the request")
+            )
+        }
+      }
+    }
+    "validation when" - {
+      "authenticated should" - {
+        "validate correct node without errors" in {
+          createSavedProcess(process, TestCategories.Category1, TestProcessingTypes.Streaming)
+          val data: node.Filter = node.Filter("id", Expression.spel("#longValue > 1"))
+          val request = NodeValidationRequest(
+            data,
+            ProcessProperties(StreamMetaData()),
+            Map("existButString" -> Typed[String], "longValue" -> Typed[Long]),
+            None,
+            None
+          )
+
+          sendRequestAsJsonAsAllpermuser(request)
+            .when()
+            .post(s"$nuDesignerHttpAddress/api/nodes/${process.id}/validation")
+            .Then()
+            .statusCode(200)
+            .body("validationErrors[0]", equalTo(null))
```

Comment:
I'm not sure, but doesn't the test pass when there was more than one `validationErrors` and the first is null? Using JSON path (because it's a json path string, right?), in general, isn't enough to check the whole JSON response